### ISSUE TITLE
Hardware configuration for running YoloV5x model

### ISSUE THREAD

**Issue by Sreekanth-aa** [ultralytics/yolov5#7387]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi,

We have used Yolov5X for object detection. The training and detection is working in Google Colab, now we want to run the detection part in local machine.

Would like to know what is the hardware configuration  (System hardware) required for running the Yolov5X in local machine.

Thanks,
Sreekanth

**Comment by Sreekanth-aa** [comment]
Hello @glenn-jocher 

Do you have any idea on this?

**Comment by glenn-jocher** [comment]
@Sreekanth-aa YOLO is largely hardware agnostic. There are no requirements.

**Comment by Sreekanth-aa** [comment]
@glenn-jocher We are running the model in a local machine with GT 730 GPU and the time taken for prediction is more. Also we testes the same weights in another PC which has GTX 1050 Ti GPU and the prediction time is less. So was thinking whether the any  specific hardware configuration is required for running the model in local machine.

**Comment by livan3li** [comment]
@glenn-jocher I'm having trouble for same issue. I've used 10k images for just training, 6 different classes and yolov5l6 Pretrained Checkpoint for training. 

and I passed arguments as 50 epochs, 40 batch size. During the training free colab session system shut down at epoch 43 and i resumed on another system and set batch size as 38. 

After that i've tried to real time inference through webcam on two different GPUs those were GeForce 940mx and GTX 1050. 

Results for both cases aren't good. Live video is freezing. 

But i've used yolov5m Pretrained Checkpoint and parameters as 50 epoch, 64 batch size for training. After this i tried on GeForce 940mx GPU and real time inference is good in terms of FPS 

so here is the question: Is there any problem with my system? or Isn't GPUs enough for real time inference? or something else.

**Comment by glenn-jocher** [comment]
@livan3li results will obviously vary by model size and deployed hardware, so before training models just run the official models on your deployment hardware to gauge which model provides sufficient speed, then train that size model.

Also if you are deploying to CUDA devices TensorRT export will provide significant speedup. No reason to run PyTorch models on GPUs in deployment. PyTorch is for training and testing mainly, i.e.:


Run YOLOv5 benchmarks (speed and accuracy) for all supported export formats. This PR adds GPU benchmarking capability following CPU benchmarking PR https://github.com/ultralytics/yolov5/pull/6613.

Format                      | `export.py --include`         | Model
:---                        | --:                           | :--
[PyTorch](https://pytorch.org/)                     | -                             | `yolov5s.pt`
[TorchScript](https://pytorch.org/docs/stable/jit.html)                 | `torchscript`                 | `yolov5s.torchscript`
[ONNX](https://onnx.ai/)                        | `onnx`                        | `yolov5s.onnx`
[OpenVINO](https://docs.openvino.ai/latest/index.html)                    | `openvino`                    | `yolov5s_openvino_model/`
[TensorRT](https://developer.nvidia.com/tensorrt)                    | `engine`                      | `yolov5s.engine`
[CoreML](https://github.com/apple/coremltools)                      | `coreml`                      | `yolov5s.mlmodel`
[TensorFlow SavedModel](https://www.tensorflow.org/guide/saved_model)       | `saved_model`                 | `yolov5s_saved_model/`
[TensorFlow GraphDef](https://www.tensorflow.org/api_docs/python/tf/Graph)         | `pb`                          | `yolov5s.pb`
[TensorFlow Lite](https://www.tensorflow.org/lite)             | `tflite`                      | `yolov5s.tflite`
[TensorFlow Edge TPU](https://coral.ai/docs/edgetpu/models-intro/)         | `edgetpu`                     | `yolov5s_edgetpu.tflite`
[TensorFlow.js](https://www.tensorflow.org/js)               | `tfjs`                        | `yolov5s_web_model/`

Usage: 
```python
git clone https://github.com/ultralytics/yolov5 -b update/bench_gpu  # clone
cd yolov5
pip install -qr requirements.txt coremltools onnx onnxruntime-gpu openvino-dev  # install
pip install -U nvidia-tensorrt --index-url https://pypi.ngc.nvidia.com  # TensorRT

python utils/benchmarks.py --weights yolov5s.pt --img 640 --device 0
```

### Colab Pro++ V100 High-RAM Results

```
benchmarks: weights=/content/yolov5/yolov5s.pt, imgsz=640, batch_size=1, data=/content/yolov5/data/coco128.yaml, device=0, half=False, test=False
Checking setup...
YOLOv5 🚀 v6.1-135-g7926afc torch 1.10.0+cu111 CUDA:0 (Tesla V100-SXM2-16GB, 16160MiB)
Setup complete ✅ (8 CPUs, 51.0 GB RAM, 46.7/166.8 GB disk)

Benchmarks complete (458.07s)
                   Format  mAP@0.5:0.95  Inference time (ms)
0                 PyTorch        0.4623                10.19
1             TorchScript        0.4623                 6.85
2                    ONNX        0.4623                14.63
3                OpenVINO           NaN                  NaN
4                TensorRT        0.4617                 1.89
5                  CoreML           NaN                  NaN
6   TensorFlow SavedModel        0.4623                21.28
7     TensorFlow GraphDef        0.4623                21.22
8         TensorFlow Lite           NaN                  NaN
9     TensorFlow Edge TPU           NaN                  NaN
10          TensorFlow.js           NaN                  NaN
```

### Colab Free T4 Results

```
benchmarks: weights=/content/yolov5/yolov5s.pt, imgsz=640, batch_size=1, data=/content/yolov5/data/coco128.yaml, device=, half=False, test=False
Checking setup...
YOLOv5 🚀 v6.1-135-g7926afc torch 1.10.0+cu111 CUDA:0 (Tesla T4, 15110MiB)
Setup complete ✅ (2 CPUs, 12.7 GB RAM, 45.2/78.2 GB disk)

Benchmarks complete (501.61s)
                   Format  mAP@0.5:0.95  Inference time (ms)
0                 PyTorch        0.4623                11.51
1             TorchScript        0.4623                 8.70
2                    ONNX        0.4623                17.08
3                OpenVINO           NaN                  NaN
4                TensorRT        0.4628                 3.36
5                  CoreML           NaN                  NaN
6   TensorFlow SavedModel        0.4623                25.13
7     TensorFlow GraphDef        0.4623                25.20
8         TensorFlow Lite           NaN                  NaN
9     TensorFlow Edge TPU           NaN                  NaN
10          TensorFlow.js           NaN                  NaN
```

Note: TensorRT exports are fixed at FP16.

**Comment by livan3li** [comment]
@glenn-jocher thank you very much for help. Im going to try as you suggest.

**Comment by glenn-jocher** [comment]
@livan3li you're welcome! If you have further questions, feel free to ask. Good luck with your experiments!
